packages/formation/js/back-to-top.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * This button is used for navigating the user back to the top of the page.
+ * It's to be used on long article pages.
+ * There is some complexity in the animation toggling function "scrollListener"
+ */
+
+function navigateToTop() {
+  const el = document.querySelector('#content');
+  if (el) el.focus();
+
+  return window.scrollTo(0, 0);
+}
+
+function isScrolledIntoView(el) {
+  const elemTop = el.getBoundingClientRect().top;
+  // Only partially || completely visible elements return true
+  return elemTop >= 0 && elemTop <= window.innerHeight;
+}
```

Comment:
Because this is triggered _every_ scroll event (which happens notoriously frequently) on _every_ page, have you done any performance benchmarking on this function? The rest of the event listener looks as fast as I can think to make it, but I'm not sure about `el.getBoundingClientRect()`.

How does it react when you use your dev tools to [throttle the CPU way down](https://umaar.com/dev-tips/88-cpu-throttling/)?

Reply:
I haven't done any benchmarking. One thing that I was interested in using was [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), but I would have had to add polyfills for IE and felt like it just wasn't worth it. However, I think that would improve performance.

Reply:
Okay, cool. It seemed pretty safe, but I wanted to verify. :+1:

Reply:
MDN recommends throttling using `window.requestAnimationFrame`, https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event